src/coreclr/jit/valuenum.cpp

```diff
@@ -9584,7 +9598,25 @@ ValueNum ValueNumStore::VNForBitCast(ValueNum srcVN, var_types castToType)
         return VNZeroForType(castToType);
     }
 
-    return VNForFunc(castToType, VNF_BitCast, srcVN, VNForIntCon(castToType));
+    ValueNum srcExcVN;
+    ValueNum srcNormVN;
+    VNUnpackExc(srcVN, &srcNormVN, &srcExcVN);
+
+    ValueNum resultNormVN = VNForFunc(castToType, VNF_BitCast, srcNormVN, VNForIntCon(castToType));
+    ValueNum resultExcVN  = srcExcVN;
+    
+    return VNWithExc(resultNormVN, resultExcVN);
+}
+
+// Compute the ValueNumberPair for a bitcast operation
```

Comment:
If we're going to add comments, they should be full header-style ones.

Reply:
Fixed.